src/commands/infra/create.ts

```diff
@@ -0,0 +1,136 @@
+      "git@github.com:timfpark/fabrikate-cloud-native-manifests.git"
+    )
+    .option(
+      "--serviceprincipalid <sp-id>",
+      "Service Principal ID for Azure Subscription"
+    )
+    .option(
+      "--serviceprincipalsecret <sp-secret> ",
+      "Service Principal Secret for Azure Subscription"
+    )
+    .action(async opts => {
+      console.log(opts.environment);
+      try {
+        if (
+          opts.environment &&
+          opts.serviceprincipalid &&
+          opts.serviceprincipalsecret
+        ) {
+        } else {
+          logger.warn(
+            "You need to specify each of the config settings in order to run any command. Please verify you have passed an Environment, Service Principal ID, and Service Principal Secret"
+          );
+        }
+        // If Bedrock is not found, install a working path for Bedrock
+        if (!fs.existsSync(bedrockDir)) {
+          const cloneBedrock = await exec("git", [
+            "clone",
+            "https://github.com/microsoft/bedrock.git",
+            ".bedrock"
+          ]);
+        }
+        await validateInit(bedrockDir);
+        await templateInit(bedrockDir, opts.environment);
+        //await templateConfig();
+        //await templateDeploy();
+      } catch (err) {
+        logger.error("Error occurred while initializing");
+        logger.error(err);
+      }
+    });
+};
+
+export const validateInit = async (bedrockPath: string): Promise<boolean> => {
+  try {
+    // TODO: Use this function to check the state of spk infra init and attain bedrock source location
+    if (fs.existsSync(bedrockPath)) {
+      if (InitIsDone) {
+        logger.info(
+          "`spk infra init` has been successfully executed, you may now proceed to deploy Bedrock environments."
+        );
+        return true;
+        logger.info(bedrockPath);
+        process.chdir(bedrockPath);
+      } else {
+        logger.error(
+          "`spk infra init` has not been successfully executed, please run this command to assure all Bedrock prerequisites are installed. "
+        );
+        return false;
+      }
+    } else {
+      logger.error(
+        `Provided Bedrock path is invalid or can not be found: ${bedrockPath}`
+      );
+      return false;
+    }
+  } catch (_) {
+    logger.error(`Unable to Validate Infra Init.`);
+    return false;
+  }
+};
+
+export const templateInit = async (
+  bedrockPath: string,
+  templateEnvironment: string
+): Promise<string> => {
+  try {
+    // Identify which environment the user selected
+    const EnvironmentPath = path.join(bedrockPath, templateEnvironment);
+    if (fs.existsSync(EnvironmentPath)) {
+      logger.info(
+        `Initializing Bedrock Template Environment : ${templateEnvironment}`
+      );
+      logger.info(EnvironmentPath);
+      process.chdir(EnvironmentPath);
+      // Terraform init in environment directory
+      const init = await exec("terraform", ["init"]);
```

Comment:
Seems like we have a need to encapsulate `terraform`. I'm thinking we could have a `terraform` specific interface that essentially acts like a wrapper or shim for all terraform commands. I think this will help with testing a lot.

Reply:
Agreed, for the second iteration, a terraform wrapper would configure the backend state with a `terraform init` and allow for more customization with an environment template for `terraform apply`.